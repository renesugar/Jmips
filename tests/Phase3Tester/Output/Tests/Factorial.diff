--- SelfTestCases/Factorial.out	2019-02-02 14:36:13.000000000 -0800
+++ ./Output/Tests/Factorial.out	2019-03-11 00:03:08.000000000 -0700
@@ -1 +1 @@
-3628800
+0
