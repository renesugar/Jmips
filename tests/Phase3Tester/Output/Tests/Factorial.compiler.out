const vmt_Fac
  :Fac.ComputeFac

func Main [in 0, out 0, local 0]
  $t8 = HeapAllocZ(4)
  [$t8] = :vmt_Fac
  if $t8 goto :null1
  Error("null pointer")
null1:
  $t7 = [$t8]
  $t7 = [$t7]
  $a0 = $t8
  $a1 = 10
  call $t7
  $t6 = $v0
  PrintIntS($t6)
  ret

func Fac.ComputeFac [in 0, out 0, local 0]
  $t6 = $a0
  $t7 = $a1
  $t8 = LtS($t7 1)
  if0 $t8 goto :if1_else
  $t8 = 1
  goto :if1_end
if1_else:
  $t5 = [$t6]
  $t5 = [$t5]
  $t4 = Sub($t7 1)
  $a0 = $t6
  $a1 = $t4
  call $t5
  $t3 = $v0
  $t8 = MulS($t7 $t3)
if1_end:
  $v0 = $t8
  ret


